twine/package.py

```diff
@@ -61,33 +61,13 @@ def __init__(self, filename, comment, metadata, python_version, filetype):
         self.signed_basefilename = self.basefilename + '.asc'
         self.gpg_signature = None
 
-        blake2_256_hash = None
-        if blake2b is not None:
-            blake2_256_hash = blake2b(digest_size=256 // 8)
-        # NOTE(sigmavirus24): We may or may not be able to use blake2 so let's
-        # either use the methods or lambdas to do nothing.
-        blake_update = getattr(blake2_256_hash, 'update', lambda *args: None)
-        blake_hexdigest = getattr(blake2_256_hash, 'hexdigest', lambda: None)
-
-        # NOTE: MD5 is not available on FIPS-enabled systems
-        md5_hash = None
-        try:
-            md5_hash = hashlib.md5()
-        except ValueError:
-            md5_hash = None
-        md5_update = getattr(md5_hash, 'update', lambda *args: None)
-        md5_hexdigest = getattr(md5_hash, 'hexdigest', lambda: None)
-
-        sha2_hash = hashlib.sha256()
-        with open(filename, "rb") as fp:
-            for content in iter(lambda: fp.read(io.DEFAULT_BUFFER_SIZE), b''):
-                md5_update(content)
-                sha2_hash.update(content)
-                blake_update(content)
+        hasher = HashManager(filename)
+        hasher.hash()
+        hexdigest = hasher.hexdigest()
```

Comment:
We could also return a namedtuple here instead of a dictionary. I didn't think it mattered much either way right now

Reply:
I definitely prefer namedtuples over dicts for cases like this.

Reply:
:heavy_check_mark: Done